### ISSUE TITLE
If some linters don't work for you (e.g. Elm), see this

### ISSUE THREAD

**Issue by megapctr** [w0rp/ale#269]
_It's not an issue per se, but I wanted to put this information somewhere_

**TL;DR**: make sure your `TMPDIR` environment variable is set!

My system: ubuntu server 16.04, fish shell 2.4.0, vim 8.0

Hi! My experience with ale has been fantastic so far. For some reason though, I couldn't get the `elm-make` linter [(link)](https://github.com/w0rp/ale/blob/master/ale_linters/elm/make.vim) to work.

After some debugging, it turned out everything works fine on the linter's side, but the `l:file_is_buffer` variable was always false, causing all errors to be skipped. As it turns out, the `TMPDIR` environment variable is not set on my system. I'm not sure whether it's because of the system or my shell, but setting this variable fixed my issues and now I can enjoy ale entirely :tada:

If you're using fish, add this to your `config.fish`
```fish
if not set -q TMPDIR
    set -gx TMPDIR /tmp
end
```

**Comment by jweir** [comment]
Ran into this same issue on Ubuntu 14.04.  I am afraid of fish and just use bash, so

```
export TMPDIR=/tmp
```

It appears that Ubuntu does not include this env var by default.  Since there is no promise this will be set maybe default to '/tmp' if TMPDIR is not defined?

**Comment by w0rp** [comment]
I wonder, could we set `TMPDIR` automatically if it is not set?

```vim
if has('unix') && empty($TMPDIR)
    let $TMPDIR = '/tmp'
endif
```

I believe any environment variables set in Vim are not exported outside of Vim, but are used for any commands launched from Vim.
